### PR TITLE
feat: integrate resend verification email API

### PR DESCRIPTION
**Pull request by YounixM**
Screenshot 2024-10-23 at 12 17 56 AM

Screenshot 2024-10-23 at 12 31 01 AM

Screenshot 2024-10-23 at 12 29 55 AM

### CODE REVIEW

**Review by therealpandey — APPROVED** [review]

File: app/verify-email/page.tsx [hunk 1]
```diff
@@ -1,67 +1,116 @@
...
-      <div className="text-md mt-[28px] rounded-[6px] border border-[#1D212D] bg-signoz_ink-300 p-[24px]">
-        <div>
-          {' '}
-          Thank you for signing up for SigNoz Cloud. Please check your email for the next steps.{' '}
+    <div className="welcome-container mx-auto flex min-h-[96vh] max-w-[520px] flex-col items-center justify-center py-32">
+      <Mail className="text-signoz_robin-500" size={56} />
+      <div className="mt-[28px] text-xl"> First, let's verify your email </div>
+
+      <div className="mt-[28px] rounded-[6px] border border-[#1D212D] bg-signoz_ink-300 p-[24px] text-sm">
+        Check your email for the next steps. If you don't see the email, check your spam folder or
```
Comment (sharpshooter90) [inline app/verify-email/page.tsx:72]: please use locale text support for the strings

File: app/verify-email/page.tsx [hunk 2]
```diff
@@ -1,67 +1,116 @@
...
+        try resending the verification email.
+      </div>
+
+      {submitFailed && (
+        <div className="mt-[28px] flex w-full items-center justify-center gap-2 text-center text-xs text-signoz_cherry-500">
+          <Frown size={16} /> It looks like something didn't go as planned. Please reach out to us
```
Comment (sharpshooter90) [inline app/verify-email/page.tsx]: please use locale text support for the strings

File: app/verify-email/page.tsx [hunk 3]
```diff
@@ -1,67 +1,116 @@
...
+          for assistance.
         </div>
+      )}
 
-        <div className="mt-[28px]">
-          If you do not find the email in your inbox within the next few minutes, please check your
-          spam folder.
+      {submitSuccess && (
+        <div className="mt-[28px] flex w-full items-center justify-center gap-2 text-center text-xs text-signoz_forest-500">
+          <CheckCircleIcon size={16} /> Verification email sent! Please check your email for the
```
Comment (sharpshooter90) [inline app/verify-email/page.tsx]: @YounixM this submit is the re-send button submit right? 

for resend use this content: 

`"Verification email resent! Please check your inbox (and spam folder) for the new verification link."`

File: app/verify-email/page.tsx (hunk 2 above)
Comment (sharpshooter90) [reply app/verify-email/page.tsx]: @YounixM for resend action failed UX copy

`"Resend failed. Please check your internet connection or try again later. If the issue persists, contact support for assistance."`

Comment (YounixM) [reply app/verify-email/page.tsx]: We don't have the locale text support in signoz web repo @sharpshooter90 .

Updating the copies provided.

Comment (YounixM) [reply app/verify-email/page.tsx]: Updated ✅

File: app/verify-email/page.tsx (hunk 3 above)
Comment (YounixM) [reply app/verify-email/page.tsx]: Updated ✅

**Review by sharpshooter90 — APPROVED** [review]

File: app/verify-email/page.tsx [hunk 4]
```diff
@@ -1,67 +1,116 @@
...
+function VerifyEmail() {
+  const [isSubmitting, setIsSubmitting] = useState(false)
+  const [submitSuccess, setSubmitSuccess] = useState(false)
+  const [submitFailed, setSubmitFailed] = useState(false)
+
+  const [workEmail, setWorkEmail] = useState<string | null>(null)
+
+  useEffect(() => {
+    try {
+      localStorage.getItem('workEmail')
```
Comment (vikrantgupta25) [inline app/verify-email/page.tsx]: this is being not assigned to something ?

File: app/verify-email/page.tsx [hunk 5]
```diff
@@ -1,67 +1,116 @@
...
+
+      if (response.ok) {
+        setSubmitSuccess(true)
+
+        setTimeout(() => {
+          setSubmitSuccess(false)
+          setSubmitFailed(false)
+        }, 5000)
+      } else {
+        // To do, handle other errors apart from invalid email
```
Comment (vikrantgupta25) [inline app/verify-email/page.tsx]: todo ?

File: app/verify-email/page.tsx [hunk 6]
```diff
@@ -1,67 +1,116 @@
...
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(payload),
+      })
+
+      if (response.ok) {
+        setSubmitSuccess(true)
+
+        setTimeout(() => {
+          setSubmitSuccess(false)
```
Comment (vikrantgupta25) [inline app/verify-email/page.tsx:51]: why this 5 seconds delay ?

Comment (YounixM) [reply app/verify-email/page.tsx:51]: Showing the message for 5 seconds and then enabling the button after 5 secs

File: app/verify-email/page.tsx (hunk 4 above)
Comment (YounixM) [reply app/verify-email/page.tsx]: Addressed ✅

File: app/verify-email/page.tsx (hunk 5 above)
Comment (YounixM) [reply app/verify-email/page.tsx]: Addressed ✅  - Removed ✅

**Review by vikrantgupta25 — APPROVED** [review]

**Pull request merged**